opentelemetry-sdk/tests/metrics/export/test_export.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import io
+import sys
+import unittest
+
+from opentelemetry.sdk import metrics
+from opentelemetry.sdk.metrics.export import ConsoleMetricsExporter
+
+
+class TestConsoleMetricsExporter(unittest.TestCase):
+    def test_export(self):
+        exporter = ConsoleMetricsExporter()
+        metric = metrics.Counter(
+            "available memory",
+            "available memory",
+            "bytes",
+            int,
+            ("environment",),
+        )
+        label_values = ("staging",)
+        handle = metric.get_handle(label_values)
+        result = '{}(data="{}", label_values="{}", metric_data={})\n'.format(
+            ConsoleMetricsExporter.__name__, metric, label_values, handle
+        )
+        output = io.StringIO()
+        sys.stdout = output
+        exporter.export([(metric, label_values)])
+        sys.stdout = sys.__stdout__
+        self.assertEqual(len(output.getvalue()), len(result))
```

Comment:
I'm not sure what kind of isolation pytest gives us, but replacing the global `stdout` here may mean that we can't run this in parallel with other tests, or that `stdout` will stay broken if `exporter.export` crashes.

We haven't paid much attention to test side effects yet (see all the tests that modify the context), but this might bite us in the future if we don't.

Consider mocking `stdout` here instead:

```python
        result = '{}(data="{}", label_values="{}", metric_data={})'.format(
            ConsoleMetricsExporter.__name__, metric, label_values, handle
        )
        with mock.patch("sys.stdout") as mock_stdout:
            exporter.export([(metric, label_values)])
        mock_stdout.write.assert_any_call(result)
```